Library/Homebrew/cask/lib/hbc/quarantine.rb

```diff
@@ -12,10 +12,28 @@ def swift
       @swift ||= DevelopmentTools.locate("swift")
     end
 
-    def available?
-      status = !swift.nil?
-      odebug "Quarantine is #{status ? "available" : "not available"}."
-      status
+    def available?(command: SystemCommand)
+      # We use Strings to force caching of the availability status.
+      @status ||= begin
+        if swift.nil?
+          odebug "Swift is not available on this system."
+          "N/A"
```

Comment:
On first read, I ask: Why `"N/A"` here instead of "`false`"? 

Or, a symbol like `:no_swift` and the output of the method is trinary: `:no_swift`, `:no_quarantine`, `:quarantine_available` or something like that, with the method called something like `check_quarantine_availability` and `available?` being a check of `check_quarantine_availability == :quarantine_available`?

But then I see: 

> \# We use Strings to force caching of the availability status.

Ah, I've not heard of this. Why is this?

Reply:
Didn't know about the symbol stuff, and it sounds way cleaner! I'll redo this.

Reply:
If you want to cache a boolean, you should use 

```ruby
return @status if defined?(@status)
```